test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -154,6 +169,39 @@ def check_output(self):
       "hwaccel initialisation returned error", self.output, re.MULTILINE)
     assert m is None, "Failed to use hardware decode"
 
+  # scale scaled test stream to source file width and height for generating yuv
+  def scale_to_source_size(self, scaled, codec, mode, width, height, frames, enc_file):
+    _, _, ffdecparms = self.get_requirements_data("decode", codec, mode)
+    ffdeccodec, _ = ffdecparms
+    _, _, ffencparms = self.get_requirements_data("encode", codec, mode)
+    ffenccodec, ext = ffencparms
+    # TBD - for mjpeg and mpeg2, HW scaling not working so using SW scaling
+    # Remove this once HW scaling works for jpeg and mjpeg2
+    if ext == "mjpeg":
+      opts = " -c:v mjpeg"
+    elif ext == "m2v":
+      opts = " -c:v mpeg2video"
+    else:
+      opts = " -hwaccel qsv -hwaccel_device /dev/dri/renderD128 -init_hw_device qsv=hw -c:v {} ".format(ffdeccodec)
+
+    opts += " -i {}".format(scaled)
+
+    if ext == "mjpeg":
+      opts += " -c:v mjpeg"
+      opts += " -vf scale=w={}:h={}".format(width, height)
+    elif ext == "m2v":
+      opts += " -c:v mpeg2video"
+      opts += " -vf scale=w={}:h={}".format(width, height)
+    else:
+      opts += " -c:v {}".format(ffenccodec)
+      opts += " -vf vpp_qsv=w={}:h={}".format(width, height)
+
+    opts += " -vframes {}".format(frames)
+    encoded = enc_file + "_ref.{}".format(ext)
```

Comment:
All artifacts should be registered with `get_media().test_artifact()` so it is properly managed.  It looks like `enc_file` is registered before calling this method.  But `enc_file + "_ref.{}` is not.

Reply:
this code removed as part other review comment to generate yuv directly from output file